activerecord/lib/active_record/query_cache.rb

```diff
@@ -27,14 +27,14 @@ def uncached(&block)
 
     def self.run
       pools = []
-      pools.concat(ActiveRecord::Base.connection_handler.connection_pool_list(:all).reject { |p| p.query_cache_enabled }.each { |p| p.enable_query_cache! })
+      pools.concat(ActiveRecord::Base.connection_handler.each_connection_pool.reject { |p| p.query_cache_enabled }.each { |p| p.enable_query_cache! })
```

Comment:
(non-blocking) we could probably reduce one more allocation, by directly setting `pools = ActiveRecord::Base.connection_handler.each_connection_pool.reject ...`. I don't think we need concat, since reject will always return a new array and this is the only thing we're concatenating into it.